TST: test_find_invalid throws TimeoutError instead of URLError now

This now times out instead of throwing URLError.

```
______________________________ test_find_invalid _______________________________

    @pytest.mark.remote_data(source="astropy")
    def test_find_invalid():
        # this is of course not a real data file and not on any remote server, but
        # it should *try* to go to the remote server
        with pytest.raises(urllib.error.URLError):
>           get_pkg_data_filename(
                "kjfrhgjklahgiulrhgiuraehgiurhgiuhreglhurieghruelighiuerahiulruli"
            )

astropy/utils/tests/test_data.py:909: 
...
E               TimeoutError: timed out

python3.11/socket.py:706: TimeoutError
```

Maybe just a matter of adjusting the test?

--- a/astropy/utils/tests/test_data.py
+++ b/astropy/utils/tests/test_data.py
@@ -905,7 +905,7 @@ def test_find_by_hash(valid_urls, temp_cache):
 def test_find_invalid():
     # this is of course not a real data file and not on any remote server, but
     # it should *try* to go to the remote server
-    with pytest.raises(urllib.error.URLError):
+    with pytest.raises((urllib.error.URLError, TimeoutError)):
         get_pkg_data_filename(
             "kjfrhgjklahgiulrhgiuraehgiurhgiuhreglhurieghruelighiuerahiulruli"
         )

